packages/suite/src/actions/wallet/moveLabelsForRbf.ts

```diff
@@ -0,0 +1,154 @@
+    {
+        toBeMoved: WalletAccountTransaction;
+        toBeDeleted: WalletAccountTransaction[];
+    }
+>;
+
+export const findLabelsToBeMovedOrDeleted = ({
+    prevTxid,
+    getState,
+}: FindLabelsToBeMovedOrDeletedParams): LabelsToBeMovedOrDeleted => {
+    const result: LabelsToBeMovedOrDeleted = {};
+
+    const accountTransactions = findTransactions(
+        prevTxid,
+        getState().wallet.transactions.transactions,
+    );
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const accountTransaction of accountTransactions) {
+        const chainedTransactionsToDrop = findChainedTransactions(
+            accountTransaction.tx.descriptor,
+            accountTransaction.tx.txid,
+            getState().wallet.transactions.transactions,
+        );
+
+        const allAccountsTransactionsIncludingChained: WalletAccountTransaction[] = [
+            accountTransaction.tx,
+            ...(chainedTransactionsToDrop?.own ?? []),
+            // Intentionally using `chainedTransactionsToDrop?.others`, they will be found when we query another account in the loop
+        ];
+
+        result[accountTransaction.key] = {
+            toBeDeleted: allAccountsTransactionsIncludingChained,
+            toBeMoved: accountTransaction.tx,
+        };
+    }
+
+    return result;
+};
+
+type MoveLabelsForRbfParams = {
+    newTxid: string;
+    toBeMovedOrDeletedList: LabelsToBeMovedOrDeleted;
+};
+
+export const moveLabelsForRbf =
+    ({ toBeMovedOrDeletedList, newTxid }: MoveLabelsForRbfParams) =>
+    async (dispatch: Dispatch, getState: GetState) => {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const toBeMovedOrDeleted of Object.entries(toBeMovedOrDeletedList)) {
+            const [accountKey, data] = toBeMovedOrDeleted;
+
+            const accountMetadata = selectLabelingDataForAccount(getState(), accountKey);
+            const accountOutputLabelsToBeMoved: AccountOutputLabels =
+                accountMetadata?.outputLabels?.[data.toBeMoved.txid] ?? {};
+
+            // eslint-disable-next-line no-await-in-loop
+            await moveLabelToNewTransaction({
+                accountKey,
+                accountOutputLabels: accountOutputLabelsToBeMoved,
+                newTxid,
+                dispatch,
+            });
+
+            // eslint-disable-next-line no-restricted-syntax
+            for (const transactionToDrop of data.toBeDeleted) {
+                const accountOutputLabelsToBeDeleted: AccountOutputLabels =
+                    accountMetadata?.outputLabels?.[transactionToDrop.txid] ?? {};
+
+                const deleteParams: DeleteAllOutputLabelsParams = {
+                    accountKey,
+                    dispatch,
+                    labels: accountOutputLabelsToBeDeleted,
+                    txid: transactionToDrop.txid,
+                };
+
+                // eslint-disable-next-line no-await-in-loop
+                await deleteAllOutputLabels(deleteParams);
```

Comment:
nit: consider renaming the action, that action is deleting unused labels, "all output" sounds too extreme :D

Reply:
:heavy_check_mark: